cmd/terramate/cli/script_info.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package cli
+
+import (
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/terramate-io/terramate/config"
+	"github.com/terramate-io/terramate/hcl"
+	"github.com/terramate-io/terramate/hcl/ast"
+)
+
+func (c *cli) printScriptInfo() {
+	labels := c.parsedArgs.Experimental.Script.Info.Labels
+
+	stacks, err := c.computeSelectedStacks(false)
+	if err != nil {
+		fatal(err, "computing selected stacks")
+	}
+
+	m := newScriptsMatcher(labels)
+	m.Search(c.cfg().Tree(), &stacks)
```

Comment:
I wonder if this pointer usage needs to be leaked to the caller.
Because the scoped stacks computation is complex and depends on flags, etc, so ideally we must do it once per Terramate invocation, but this method is destructive and `stacks` cannot be used afterward.

Maybe you can make `Search()` receive the list (not a pointer) and then inside it yo add:
```
copied := make(..., len(stacks))
copy(copied, stacks)
m.searchRecursive(cfg, &copied)
```
then just the `copied` is mutated.

Additionally, it looks like only the stack paths are used, then maybe just passing `stacks.Paths()` would suffice?

Reply:
Good point, I changed it as you suggested. I couldnt just use the paths because it looks to me like script run needs  them.